core/services/llo/delegate.go

```diff
@@ -0,0 +1,119 @@
+package llo
+
+import (
+	"context"
+	"fmt"
+
+	ocrcommontypes "github.com/smartcontractkit/libocr/commontypes"
+	ocr2types "github.com/smartcontractkit/libocr/offchainreporting2/types"
+	ocr2plus "github.com/smartcontractkit/libocr/offchainreporting2plus"
+	ocr3types "github.com/smartcontractkit/libocr/offchainreporting2plus/ocr3types"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+	commontypes "github.com/smartcontractkit/chainlink-common/pkg/types"
+	"github.com/smartcontractkit/chainlink-data-streams/llo"
+
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/job"
+	"github.com/smartcontractkit/chainlink/v2/core/services/llo/evm"
+	"github.com/smartcontractkit/chainlink/v2/core/services/pg"
+	"github.com/smartcontractkit/chainlink/v2/core/services/streams"
+)
+
+var _ job.ServiceCtx = &delegate{}
+
+type Closer interface {
+	Close() error
+}
+
+type delegate struct {
+	services.StateMachine
+
+	cfg    DelegateConfig
+	codecs map[commontypes.LLOReportFormat]llo.ReportCodec
+
+	oracle Closer
+}
+
+type DelegateConfig struct {
+	Logger   logger.Logger
+	Queryer  pg.Queryer
+	Runner   streams.Runner
+	Registry Registry
+
+	// LLO
+	ChannelDefinitionCache commontypes.ChannelDefinitionCache
+
+	// OCR3
+	BinaryNetworkEndpointFactory ocr2types.BinaryNetworkEndpointFactory
+	V2Bootstrappers              []ocrcommontypes.BootstrapperLocator
+	ContractConfigTracker        ocr2types.ContractConfigTracker
```

Comment:
mixing ocr2types and ocr3types. can we just use ocr3types everywhere?

Reply:
We can use `ocr2plus` instead of `ocr2` here, good catch :+1: